src/main/java/de/tum/in/ase/parser/strategy/GCCParser.java

```diff
@@ -0,0 +1,189 @@
+package de.tum.in.ase.parser.strategy;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import de.tum.in.ase.parser.domain.Issue;
+import de.tum.in.ase.parser.domain.Report;
+
+public class GCCParser implements ParserStrategy {
+
+    // The message is split into two segments
+    private static final int SEGMENTS_COUNT = 2;
+
+    // Segment 0 (Header): Contains info about filename, row, column, type etc.
+    private static final int HEADER_SEGMENT_POS = 0;
+
+    // Segment 1 (Body): Body contains the trace of the error (possibly rendered with ASCII art)
+    private static final int BODY_SEGMENT_POS = 1;
+
+    // Locations in regex group
+    private static final int FILE_POS = 1;
+
+    private static final int ROW_POS = 2;
+
+    private static final int COLUMN_POS = 3;
+
+    private static final int TYPE_POS = 4;
+
+    private static final int DESCRIPTION_POS = 5;
+
+    private static final int ERROR_POS = 6;
+
+    // Map that contains the matching category for each error
+    private static final Map<String, String> categories = new HashMap<>();
+
+    private static final String ANALYZER_PREFIX = "[-Wanalyzer";
+
+    // Categories
+    private static final String MEMORY = "Memory";
+
+    private static final String BAD_PRACTICE = "BadPractice";
+
+    private static final String SECURITY = "Security";
+
+    private static final String UNDEFINED_BEHAVIOR = "UndefinedBehavior";
+
+    private static final String MISC = "Misc"; // For various other results that are not part of the static analysis
```

Comment:
```suggestion
    // For various other results that are not part of the static analysis
    private static final String MISC = "Misc";
```
Nitpick: Just to match the comment style of the other declarations.

Reply:
I agree, that comment kinda looks off. I wanted to have it act like a description here, and the other comments that have their own line as "structural" comments. But it would be still kinda inconsistent. Added your suggestion :+1: